### DISCUSSION TITLE
Box with scrollbars

### DISCUSSION THREAD

**Discussion by radoslaw-sz** [xyflow/xyflow#3251] [Q&A]
Hi everyone,

TLDR:
Is it possible to create a box, which will be limited by positions of nodes and will be scrollable if does not fit a screen? I want to have a rectangle where the flow will be placed and user can just scroll (horizontally and vertically with scrollbars) to see whole flow.
At this moment, I have a flow which is scrollable, but I can indefinitely scroll, scrolling is not limited at all even if I see the last element from the flow. Additionally, I do not see scrollbars..

Long version:
Not sure if it is achievable, but my plan is to create a flow like CI pipeline. Because of it, I would like to have a flow which is horizontally scrollable, so if my elements are overflowed (on the right side), then I want to show user scrollbar and user can scroll to the right to see the latest element.
Same story for vertical - if pipeline will be branched out, then user shall be able to scroll down to see branch. 
The assumption is that the whole flow is placed inside some box (so user can scroll also outside of the flow to see rest of elements in page after a flow). 
Flow needs to be read-only.

**Comment by radoslaw-sz** [comment]
Ok, I was able to introduce limitation by playing with `translateExtent`. However, scrollbar issue left...

**Reply by bcakmakoglu** [reply]
As was mentioned in another issue, scrollbars aren't something that will be built-in.
You can calculate the width/height you need and resize your wrapping element accordingly so you can add scrollbars yourself but the viewport of React Flow is intentionally infinite and thus will not have scrollbars to it.

**Reply by radoslaw-sz** [reply]
ok, thanks for the explanation
